src/test/java/org/assertj/core/internal/files/Files_assertCanWrite_Test.java

```diff
@@ -46,18 +43,19 @@ void should_fail_if_actual_is_null() {
 
   @Test
   void should_fail_if_can_not_write() {
-    when(actual.canWrite()).thenReturn(false);
+    // GIVEN
+    File actual = new File("xyz");
     AssertionInfo info = someInfo();
-
+    // WHEN
     Throwable error = catchThrowable(() -> files.assertCanWrite(info, actual));
-
+    // THEN
     assertThat(error).isInstanceOf(AssertionError.class);
     verify(failures).failure(info, shouldBeWritable(actual));
   }
 
   @Test
   void should_pass_if_actual_can_write() {
-    when(actual.canWrite()).thenReturn(true);
+    File actual = newFile(tempDir.getAbsolutePath() + "to_write.txt");
```

Comment:
Even if it's experimental let's give a try to JUnit5 [@TempDir](https://junit.org/junit5/docs/current/user-guide/#writing-tests-built-in-extensions-TempDirectory) to avoid leaving temp files.

Have a look at `PathsBaseTest` where it is already used.

Reply:
Comment applies to other tests using temp files